pkg/binary/envoy/run.go

```diff
@@ -35,15 +35,25 @@ func (r *Runtime) Run(ctx context.Context, args []string) error {
 	cmd.Stderr = r.Err
 	cmd.SysProcAttr = sysProcAttr()
 	r.cmd = cmd
+	// suppress any error and replace it with the envoy exit status when > 1
+	defer func() {
+		if cmd.ProcessState != nil && cmd.ProcessState.ExitCode() > 0 {
+			if err != nil {
+				fmt.Fprintln(r.Out, "warning:", err)
+			}
+			err = fmt.Errorf("envoy exited with status: %d", cmd.ProcessState.ExitCode())
```

Comment:
now, we know exactly what the envoy exit code was from looking at logs. Otherwise, we don't redundantly log